netbox/core/models/files.py

```diff
@@ -104,8 +125,9 @@ def clean(self):
 
     def delete(self, *args, **kwargs):
         # Delete file from disk
+        storage = self.get_storage()
         try:
-            os.remove(self.full_path)
+            storage.delete(self.full_path)
```

Comment:
I think there's a bug in the path resolution preventing files from deleted from the S3 bucket when deleting a script module. In my testing, `full_path` returns the filename prefixed with `scripts/` twice.